sync-android/src/main/java/com/cloudant/sync/sqlite/android/AndroidSQLCipherSQLite.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) 2013 Cloudant, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.sqlite.android;
+
+/**
+ * Similar to AndroidSQLite class.  Replaces Android SQLite implementation with
+ * SQLCipher-based SQLite.
+ * SQLCipher adds:
+ * - TODO A user-based password (provided with the KeyProvider object) for encryption
+ * - TODO Method to change password (will be handled through key management feature)
+ *
+ * Created by estebanmlaver.
+ */
+
+
+import com.cloudant.sync.datastore.encryption.KeyProvider;
+import com.cloudant.sync.datastore.encryption.KeyUtils;
+import com.cloudant.sync.sqlite.ContentValues;
+import com.cloudant.sync.sqlite.Cursor;
+import com.cloudant.sync.sqlite.SQLDatabase;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+import net.sqlcipher.database.SQLiteConstraintException;
+import net.sqlcipher.database.SQLiteDatabase;
+import net.sqlcipher.database.SQLiteException;
+
+public class AndroidSQLCipherSQLite extends SQLDatabase {
+
+    SQLiteDatabase database = null;
+    private Thread threadWhichOpened = null;
+    private ThreadLocal<Boolean> appearsOpen = null;
```

Comment:
Thread Local code isn't needed here, it was removed from `AndroidSQLite` in PR #128 same should be applied here